client/state/data-layer/wpcom/timezones/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import { TIMEZONES_REQUEST } from 'state/action-types';
+
+import {
+	timezonesRequestSuccessAction,
+	timezonesRequestFailureAction,
+	timezonesReceiveAction,
+} from 'state/timezones/actions';
+
+const undocumented = wpcom.undocumented();
+
+/*
+ * Start a request to WordPress.com server to get the timezones data
+ */
+export const requestTimezones = ( { dispatch } ) => (
+	undocumented
+		.timezones()
+		.then( zones => {
+			dispatch( timezonesRequestSuccessAction() );
+			dispatch( timezonesReceiveAction( zones ) );
```

Comment:
> this is the perfect place to use fromApi() to normalize the names

👍 We should pay attention to those redux implementations which are already assembling their data gotten from the REST API though.